src/kwsbin/transcripts-to-fsts.cc

```diff
@@ -81,6 +108,15 @@ int main(int argc, char *argv[]) {
 
       VectorFst<StdArc> fst;
       MakeLinearAcceptor(transcript, &fst);
+      if (weights_rspecifier != "" ) {
+        std::string weights_key = weights_reader.Key();
+        double weight = weights_reader.Value();
+        weights_reader.Next();
+
+        KALDI_ASSERT(weights_key == key);
+
```

Comment:
This code is a bit ugly, and I think unnecessary... the weights_reader should be made a RandomAccessDoubleReader, and you can use HasKey(key) and Value(key).  ,s,cs can be added on the command line to make it efficient, if they are actually sorted.  If they are not sorted (and this might relate to the special things that are done in the keyword-search pipeline), then we'll just take the memory hit.  I prefer tools like this to be general purpose.